kernel/common/ezautolinkoperator.php

```diff
@@ -42,15 +54,52 @@ function formatUri( $url, $max )
         }
         return "<a href=\"$url\" title=\"$url\">$text</a>";
     }
+    
+    static function formatUriCallback( $matches )
+    {
+        $method = $matches[1];
+        $www = $matches[2];
+        $domainPath = $matches[3];
+        
+        $url = $method . $www . $domainPath;
+        
+        if ( self::$isLinkWwwUrl && $www == 'www.' )
+        {
+            $url = 'http://' . $www . $domainPath;
+        }
+        
+        $textFull = $url;
+        
+        if ( self::$hideProtoclText != '' )
+        {
+            $textFull = preg_replace( '`^(?:' . self::$hideProtoclText . '):\/\/(.*)$`', '$1', $textFull );
+        }
+        
+        $text = $textFull;
+        if (strlen($text) > self::$max)
+        {
+            $text = substr($text, 0, (self::$max / 2) - 3). '...'. substr($text, strlen($text) - (self::$max / 2));
+        }
+        return "<a href=\"$url\" title=\"$textFull\">$text</a>";
+    }
 
     /*!
+      $max - leave for backward compatibilty
      \static
     */
-    function addURILinks( $text, $max, $methods = 'http|https|ftp' )
+    static function addURILinks( $text, $max = false, $methods = 'http|https|ftp' )
     {
-        return preg_replace(
-            "`(?<!href=\"|href='|src=\"|src='|value=\"|value=')($methods):\/\/[\w]+(.[\w]+)([\w\-\.,@?^=%&:\/~\+#;*\(\)\!]*[\w\-\@?^=%&\/~\+#;*\(\)\!])?`e",
-            'eZAutoLinkOperator::formatUri("$0", '. $max. ')',
+        if ( $max !== false && is_int( $max ) )
+            self::$max = $max;
+        
+        if ( self::$isLinkWwwUrl )
```

Comment:
Please use if conditional statement brackets here to conform to the coding standard and increase readability.